### ISSUE TITLE
Training using Mahalanobis distance does not seem to work

### ISSUE THREAD

**Issue by BenCretois** [NINAnor/rare_species_detections#6]
When training with the Mahalanobis distance instead of the Euclidean distance it seems that the network is not training i.e. the loss is not decreasing but remain constant. 

I have tried changing the learning rate (both higher and lower LR) but it does not seem to change. 

The function may not be well implemented and the gradients might get lost somewhere in the [mahalanobis_distance](https://github.com/NINAnor/rare_species_detections/blob/94178125a89e413561b354a2485b9854a52b66a0/prototypicalbeats/prototraining.py#L61) function.
